spec/features/graph/legend_spec.rb

```diff
@@ -15,8 +15,18 @@
         expect(page).to have_content "prometheus_targetpool_duration_ms"
       end
     end
+
+    it "maintains legend state after update" do
+      expect(all('.rickshaw_legend .line.disabled').count).to eq 0
+      legend_elements = all('.rickshaw_legend .action')
+      legend_elements.first.click
+      expect(all('.rickshaw_legend .line.disabled').count).to eq 1
+      open_tab 'Refresh'
```

Comment:
In my experience we should try to avoid sleeps in tests to ensure a state change. It does not ensure that a change really has happened, slows down tests and exposes the test suite to potential one-off test errors (slow execution on CI servers, et.).

What data does the test server return? Could we implement it in a way that it's for example a monotonically increasing counter? Then we could add a method which waits/ensures that the last data point has been updated before we check the test expectation.